department-of-veterans-affairs/va.gov-cms
# Revert GHA Tugboat integration

## Description

Resolves #16884. This will revert the changes from https://github.com/department-of-veterans-affairs/va.gov-cms/pull/16279 https://github.com/department-of-veterans-affairs/va.gov-cms/pull/16562 and https://github.com/department-of-veterans-affairs/va.gov-cms/pull/16739. These changes will return to using webhook integration for Tugboat previews.

## Testing done

Webhooks tested using the following steps with a [test PR](https://github.com/department-of-veterans-affairs/va.gov-cms/pull/17123).

## QA steps

Testing this change requires turning on the Tugboat webhook which will be disruptive since duplicate Tugboat previews will be created. When testing make an announcement in the CMS channel so other developers are aware of the issue and clean up duplicated previews once testing is done. To test this change, make a PR that targets this branch and perform the following actions which tests for routine PR actions including opening, updating, closing, reopening and merging.

As any user who can open PRs on this repo
1. Turn on Tugboat webhooks in the repository settings and enable sending PR notifications
   - [ ] Navigate to Settings >> Webhooks >> https://dev.va.gov/cms-tugboat/... >> Edit
   - [ ] Under "Which events would you like to trigger this webhook?" choose "Let me select individual events."
   - [ ] Check "Pull requests" and do not uncheck "Pushes"
   - [ ] Click "Update Webhook"
2. Open a PR targeting this branch `16884-revert`. I recommend making a small documentation change that can be reverted in step 6.
   - [ ] Ensure required CI workflows are triggered and completes successfully
   - [ ] Ensure Tugboat deployments are triggered and the status and links are posted to the PR. E.g.
   
<img width="511" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/2030323/188b5af8-d497-4dba-85a8-955e6ecfbd79">
<img width="915" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/2030323/5f5bc2b1-12e2-4323-8383-dcf00763e7cb">

   - [ ] Ensure previews are created and can be accessed
3. Update the PR with an new commit
   - [ ] Ensure required CI workflows are triggered and completes successfully
   - [ ] Ensure previews are rebuilt on PR updates, ensure the SHA of the commit used to build the preview is the updated commit.
   - [ ] Ensure previews are created and can be accessed
4. Close the PR without merging
   - [ ] Ensure Tugboat deployments status is updated (i.e. deployment is destroyed). E.g:
   
<img width="535" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/2030323/84aaf5d7-312a-4fad-83bd-2c3fab8679c0">

   - [ ] Ensure previews are deleted
5. Re-open the PR
   - [ ] Ensure required CI workflows are triggered and completes successfully
   - [ ] Ensure Tugboat deployments are triggered and the status and links are posted to the PR
   - [ ] Ensure previews are created and can be accessed
5. Merge the PR to this branch
   - [ ] Ensure Tugboat deployments status is updated (i.e. deployment is destroyed)
   - [ ] Ensure previews are deleted
7. Clean up changes made during these tests
  - [ ] Revert the the changes contained in the PR so it does not affect this PR functionally.
  - [ ] Turn off the Tugboat PR webhook
  - [ ] Review Tugboat previews and delete any duplicates created during testing

### Definition of Done


### Select Team for PR review
